packages/arduino-cli/src/index.js

```diff
@@ -30,51 +37,27 @@ const ArduinoCli = (pathToBin, config = null) => {
   };
 
   const escapedConfigPath = escapeSpacesNonWin(configPath);
-  const run = args => {
-    const promise = exec(
-      `"${pathToBin}" --config-file=${escapedConfigPath} ${args}`
-    )
-      .then(
-        R.tap(() => {
-          deleteProcess(promise.childProcess);
-        })
-      )
-      .then(R.prop('stdout'));
-
-    appendProcess(promise.childProcess);
-
-    return promise;
-  };
   const runWithProgress = async (onProgress, args) => {
-    const spawnArgs = R.compose(
-      R.concat([`--config-file=${escapedConfigPath}`]),
-      R.reject(R.isEmpty),
-      R.split(' ')
-    )(args);
-
-    const promise = spawn(escapeSpacesNonWin(pathToBin), spawnArgs, {
+    const spawnArgs = R.concat([`--config-file=${escapedConfigPath}`], args);
+    const proc = spawn(escapeSpacesNonWin(pathToBin), spawnArgs, {
       stdio: ['inherit', 'pipe', 'pipe'],
-      shell: true,
+      shell: !IS_WIN, // TODO: Can we get rid of `shell: true` for other platforms?
```

Comment:
This indeed is the question. Shell features are necessary only if they are used (obviously). Do we use either of the things below?

- shell pipes
- filename wildcards
- environment variable expansion
- expansion of `~` to a user’s home directory

At first glance, we don’t. Needs testing though. Also, make sure the long-running process output is output in XOD deployment pane as it goes, that is in real time, with the setting chosen.

Reply:
`+1` here. Using shell leads to more bugs.